bftengine/src/bcstatetransfer/BCStateTran.cpp

```diff
@@ -1333,6 +1340,35 @@ bool BCStateTran::onMessage(const CheckpointSummaryMsg *m, uint32_t msgLen, uint
   return true;
 }
 
+uint16_t BCStateTran::asyncGetBlocksConcurrent(uint64_t nextBlockId,
+                                               uint64_t firstRequiredBlock,
+                                               uint16_t numBlocks,
+                                               size_t startContextIndex) {
+  ConcordAssertGE(config_.maxNumberOfChunksInBatch, numBlocks);
+  auto j{startContextIndex};
+
+  std::set<size_t> indexesToWaitfor;
+
+  for (uint64_t i{nextBlockId}; (i >= firstRequiredBlock) && (j < startContextIndex + numBlocks); --i, ++j) {
```

Comment:
I don't see how?
I added it for better readability (I guess for some people it doesn't help much..), it is a good practice in my opinion. In some cases, it can also prevent bugs in complex if statements.

Reply:
In my opinion, her specifically the condition is quite regular but if you think it may confuse someone, leave it as is.